### DISCUSSION TITLE
Unable to logout when using with Keycloak

### DISCUSSION THREAD

**Discussion by Jonty16117** [outline/outline#7860] [Self Hosting]
I am using a self hosted setup of Keycloak and Outline. Using normal email and password to login, the user get logged into the outline but when I logout from the Outline dashboard, after few refreshes, I am back to the Outline homepage again, in the keycloak dashboard as well I can see the user session is not deleted. Key point to be noted is that on clicking the logout button from the Outline dashboard, this api fails: `POST https://k8base.xxxxx.xyz/api/auth.delete 401 (Unauthorized)`

![1](https://github.com/user-attachments/assets/557cca93-dc21-447a-aca6-6398ece6393a)
![2](https://github.com/user-attachments/assets/36a6c9cf-4f0b-40f0-bc4f-7bf97d5ee006)
![3](https://github.com/user-attachments/assets/bec5ee4c-2ac2-47be-8da2-9885a14498f4)
![4](https://github.com/user-attachments/assets/c02efd93-f84b-4c49-bad3-83f1e9a71060)

This is my keycloak settings
![5](https://github.com/user-attachments/assets/aa4afbdc-97a6-4b94-bbb2-e16168119cbf)
![image](https://github.com/user-attachments/assets/955f1772-2f16-48b1-8cb5-f4067a2c7cc3)

**Comment by tommoor** [comment]
Please see the documentation here – in particular the second paragraph:
https://docs.getoutline.com/s/hosting/doc/oidc-8CPBm6uC0I

**Reply by Jonty16117** [reply]
I do have this env set, but still not working, I see some redirect glitches and I am unable to logout and the page just reloads again
![1](https://github.com/user-attachments/assets/6831d812-6674-4b0a-abc6-73f89fdbc3b7)

**Reply by tommoor** [reply]
Sure, but that should point to the logout url on keycloak – not just the keycloak instance, this is why you're in a loop

**Reply by Jonty16117** [reply]
even with `OIDC_LOGOUT_URI=https://keycloak.xxxxxxxxx.xyz/realms/outline/protocol/openid-connect/logout` path there is same issue

**Comment by andrewkcarter** [comment]
I can reproduce the same issue. Specifically, 0.78.0 logs out correctly from the auth server (keycloak), and 0.79.0 doesn't. Seems like https://github.com/outline/outline/pull/7446 has broken logout.

Here are the network calls for 0.78:
<img width="1064" alt="Screenshot 2024-11-22 at 1 41 16 PM" src="https://github.com/user-attachments/assets/b09ee7e3-26e0-468d-a80c-6ed98a8d7fac">

In .78 we see two sequential `auth.delete` calls that succeed, followed by a redirect to the OIDC_LOGOUT_URI.

Here are the network calls for 0.79 up to 0.81.1:
<img width="1061" alt="Screenshot 2024-11-22 at 1 43 12 PM" src="https://github.com/user-attachments/assets/bd0a9e0d-4e0e-4379-92b4-b05e32bbea02">

In 0.79+, the second `auth.delete` always gets a 401 and the browser is never redirected to `OIDC_LOGOUT_URI`. Instead, the user gets put through the authorization flow again, which picks up the existing user session on the auth server. The user is unable to actually logout.

My configuration is only OIDC with Keycloak, no other providers.

**Comment by andrewkcarter** [comment]
I just stumbled onto `OIDC_DISABLE_REDIRECT` and setting this to `true` allows the browser to invoke the `OIDC_LOGOUT_URI` when the user logs out. So, in short it "works" but then users are forced to click through the SSO login page, even though there is only one option.
<img width="377" alt="Screenshot 2024-11-22 at 2 02 34 PM" src="https://github.com/user-attachments/assets/1c52f5b4-d9fc-4d5c-b136-49bc2f15b50b">

Sub-optimal, and it feels like the logout logic is still not 100% correct.
